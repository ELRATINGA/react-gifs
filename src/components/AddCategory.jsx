import { useState } from "react"

export const AddCategory = ({ onNewCategorie }) => {
    
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const onSubmitNotReloy = (even) => {
        even.preventDefault();
        const cleanInput = inputValue.trim();

        if (cleanInput <= 1) return;

      //setCategories(categories => [ ...categories, inputValue ]);
      //setCategories([ ...categories, 'Dead by Deadlight' ] ); Esta solo funciona si en los props le mando 'categories'.

        onNewCategorie(cleanInput);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmitNotReloy }> {/*TambiÃ©n se puede poner en 'onSubmitNotReloy' '(even) => onSubmitNotReloy(even)'*/}

        <input 
            type="text" 
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={ onInputChange } // Otra manera de hacerlo '(evento) => onInputChange(evento)'
        />   
        
        </form>
    )
}
