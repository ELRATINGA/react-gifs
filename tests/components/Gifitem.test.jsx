import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Gifitem', () => { 
    
    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render( <GifItem key={1} title="Genshin" url="xd" id={1} /> )

        expect( container ).toMatchSnapshot();
     });

     test('Debe mostrar la imagen y el ALT indicado', () => { 
        render( <GifItem key={1} title="Genshin" url="http://localhost//xd" id={1} /> )
        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( 'http://localhost//xd' );
        expect( alt ).toBe( 'Genshin' );
      })
 })