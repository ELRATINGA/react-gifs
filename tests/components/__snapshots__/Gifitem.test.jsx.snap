// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifitem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Genshin"
      src="xd"
    />
    <p>
      Genshin
    </p>
  </div>
</div>
`;
